contracts/deployables/modules/AzoriusV1.sol

```diff
@@ -337,63 +277,28 @@ contract AzoriusV1 is IAzoriusV1, GuardableModule, Version, UUPSUpgradeable {
             uint32 _executionCounter
         )
     {
-        _strategy = proposals[_proposalId].strategy;
-        _txHashes = proposals[_proposalId].txHashes;
-        _timelockPeriod = proposals[_proposalId].timelockPeriod;
-        _executionPeriod = proposals[_proposalId].executionPeriod;
-        _executionCounter = proposals[_proposalId].executionCounter;
-    }
-
-    /** @inheritdoc IAzoriusV1*/
-    function enableStrategy(address _strategy) public override onlyOwner {
-        if (_strategy == address(0) || _strategy == SENTINEL_STRATEGY)
-            revert InvalidStrategy();
-        if (strategies[_strategy] != address(0)) revert StrategyEnabled();
-
-        strategies[_strategy] = strategies[SENTINEL_STRATEGY];
-        strategies[SENTINEL_STRATEGY] = _strategy;
-
-        emit EnabledStrategy(_strategy);
-    }
-
-    /** @inheritdoc IAzoriusV1*/
-    function disableStrategy(
-        address _prevStrategy,
-        address _strategy
-    ) public onlyOwner {
-        if (_strategy == address(0) || _strategy == SENTINEL_STRATEGY)
-            revert InvalidStrategy();
-        if (strategies[_prevStrategy] != _strategy) revert StrategyDisabled();
-
-        strategies[_prevStrategy] = strategies[_strategy];
-        strategies[_strategy] = address(0);
-
-        emit DisabledStrategy(_strategy);
-    }
-
-    /** @inheritdoc IAzoriusV1*/
-    function isStrategyEnabled(address _strategy) public view returns (bool) {
-        return
-            SENTINEL_STRATEGY != _strategy &&
-            strategies[_strategy] != address(0);
+        Proposal memory _proposal = proposals[_proposalId];
+        _strategy = _proposal.strategy;
+        _txHashes = _proposal.txHashes;
+        _timelockPeriod = _proposal.timelockPeriod;
+        _executionPeriod = _proposal.executionPeriod;
+        _executionCounter = _proposal.executionCounter;
     }
 
     /** @inheritdoc IAzoriusV1*/
     function proposalState(
         uint32 _proposalId
     ) public view returns (ProposalState) {
+        if (_proposalId >= totalProposalCount) revert InvalidProposal();
         Proposal memory _proposal = proposals[_proposalId];
 
-        if (_proposal.strategy == address(0)) revert InvalidProposal();
-
-        IBaseStrategyV1 _strategy = IBaseStrategyV1(_proposal.strategy);
-        (, uint48 votingEndTimestamp) = _strategy.getVotingTimestamps(
+        (, uint48 votingEndTimestamp) = strategy.getVotingTimestamps(
```

Comment:
Bug: need to check the _proposal's_ strategy, because the contract's strategy might have been replaced while a proposal is active.